source/server.go

```diff
@@ -31,7 +31,8 @@ import (
 type Server struct {
 	pb.UnimplementedSourceServiceServer
 
-	RecordCh chan sdk.Record
+	RecordCh      chan sdk.Record
+	StreamIsSetCh chan bool
```

Comment:
Since the boolean in the channel doesn't contain any information (it will never be `false`), we can use `chan struct{}` like for the `teardown` channel. The empty struct doesn't consume any bytes.

Btw, it doesn't seem like the field needs to be exposed, can we make it private?

Reply:
oh thought an empty struct was the same size as a boolean!